src/types/appleSk2.ts

```diff
@@ -134,8 +134,22 @@ export type SubscriptionStatus =
   | 'revoked'
   | 'subscribed';
 
+/**
+* Renewal info for whole subscription group.
+* see: https://developer.apple.com/documentation/storekit/product/subscriptioninfo/status/3822294-renewalinfo
+* WARN:
+* - autoRenewPreference is serialised as autoRenewProductId in jsonRepresentation
+* - renewalDate is available in jsonRepresentation (will change with Xcode 15 https://developer.apple.com/forums/thread/738833)
```

Comment:
once we will update to Xcode 15, then we will be able to get renewal date form new API , see Apple answer here: https://developer.apple.com/forums/thread/738833